scopes/ui-foundation/ui/ui.main.runtime.ts

```diff
@@ -462,15 +471,23 @@ export class UiMain {
   async buildUiHash(uiRoot: UIRoot, runtime = 'ui'): Promise<string> {
     const aspects = await uiRoot.resolveAspects(runtime);
     aspects.sort((a, b) => (a.aspectPath > b.aspectPath ? 1 : -1));
-    const hash = aspects.map((aspect) => {
+    const aspectPathStrings = aspects.map((aspect) => {
       return [aspect.aspectPath, aspect.runtimePath].join('');
     });
-    return sha1(hash.join(''));
+    return sha1(aspectPathStrings.join(''));
   }
 
   async buildIfChanged(name: string, uiRoot: UIRoot, force: boolean | undefined): Promise<string> {
     this.logger.debug(`buildIfChanged, name ${name}`);
     const hash = await this.buildUiHash(uiRoot);
+
+    const preBundleHash = getPreBundleHash();
+    if (hash === preBundleHash && !force) {
```

Comment:
Like on the other pr comment, i'm not sure these 2 hashes will ever match.
please validate it.